tests/component/test_task_events.py

```diff
@@ -0,0 +1,414 @@
+    )
+
+
+@pytest.mark.xfail(
+    reason="AB#2395958: GrpcStubInterpreter does not report event registration errors to caller"
+)
+def test___ao_task___register_wrong_every_n_samples_event___not_supported_by_device_error_raised(
+    ao_task: nidaqmx.Task,
+) -> None:
+    event_observer = EveryNSamplesEventObserver()
+    ao_task.timing.cfg_samp_clk_timing(
+        rate=10000.0, sample_mode=AcquisitionType.FINITE, samps_per_chan=1000
+    )
+
+    with pytest.raises(nidaqmx.DaqError) as exc_info:
+        ao_task.register_every_n_samples_acquired_into_buffer_event(
+            100, event_observer.handle_every_n_samples_event
+        )
+
+    assert (
+        exc_info.value.error_code
+        == DAQmxErrors.EVERY_N_SAMPLES_ACQ_INTO_BUFFER_EVENT_NOT_SUPPORTED_BY_DEVICE
+    )
+
+
+@pytest.mark.xfail(reason="AB#2395984: GrpcStubInterpreter does not unregister events with DAQmx")
+def test___task___register_unregister_done_event___callback_not_invoked(
+    ai_task: nidaqmx.Task,
+) -> None:
+    event_observer = DoneEventObserver()
+
+    for _ in range(10):
+        ai_task.register_done_event(event_observer.handle_done_event)
+        ai_task.register_done_event(None)
+
+    assert len(event_observer.events) == 0
+
+
+@pytest.mark.xfail(reason="AB#2395984: GrpcStubInterpreter does not unregister events with DAQmx")
+def test___task___register_unregister_every_n_samples_acquired_into_buffer_event___callback_not_invoked(
+    ai_task: nidaqmx.Task,
+) -> None:
+    event_observer = EveryNSamplesEventObserver()
+
+    for _ in range(10):
+        ai_task.register_every_n_samples_acquired_into_buffer_event(
+            100, event_observer.handle_every_n_samples_event
+        )
+        ai_task.register_every_n_samples_acquired_into_buffer_event(100, None)
+
+    assert len(event_observer.events) == 0
+
+
+@pytest.mark.xfail(reason="AB#2395984: GrpcStubInterpreter does not unregister events with DAQmx")
+def test___task___register_unregister_every_n_samples_transferred_from_buffer_event___callback_not_invoked(
+    ao_task: nidaqmx.Task,
+) -> None:
+    event_observer = EveryNSamplesEventObserver()
+
+    for _ in range(10):
+        ao_task.register_every_n_samples_transferred_from_buffer_event(
+            100, event_observer.handle_every_n_samples_event
+        )
+        ao_task.register_every_n_samples_transferred_from_buffer_event(100, None)
+
+    assert len(event_observer.events) == 0
+
+
+@pytest.mark.xfail(reason="AB#2395984: GrpcStubInterpreter does not unregister events with DAQmx")
+def test___task___register_unregister_signal_event___callback_not_invoked(
+    ai_task: nidaqmx.Task,
+) -> None:
+    event_observer = SignalEventObserver()
+
+    for _ in range(10):
+        ai_task.register_signal_event(Signal.SAMPLE_COMPLETE, event_observer.handle_signal_event)
+        ai_task.register_signal_event(Signal.SAMPLE_COMPLETE, None)
+
+    assert len(event_observer.events) == 0
```

Comment:
I was pleasantly surprised to see that these tests fail with gRPC: grpc-device reports the registration error as a normal response, so we call the event callback. For every N and signal events, we ignore the response's `status` field, though...